cms/static/js/views/components/add_library_content.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Provides utilities to open and close the library content picker.
+ *
+ */
+define(['jquery', 'underscore', 'gettext', 'js/views/modals/base_modal'],
+function($, _, gettext, BaseModal) {
+    'use strict';
+
+    var AddLibraryContent = BaseModal.extend({
+        options: $.extend({}, BaseModal.prototype.options, {
+            modalName: 'add-component-from-library',
+            modalSize: 'lg',
+            view: 'studio_view',
+            viewSpecificClasses: 'modal-add-component-picker confirm',
+            // Translators: "title" is the name of the current component being edited.
+            titleFormat: gettext('Add library content'),
+            addPrimaryActionButton: false,
+        }),
+
+        initialize: function() {
+            BaseModal.prototype.initialize.call(this);
+            // Add event listen to close picker when the iframe tells us to
+            window.addEventListener("message", function (event) {
+                if (event.data?.type === 'pickerComponentSelected') {
+                    var requestData = {
+                        library_content_key: event.data.usageKey,
+                        category: event.data.category,
+                    }
+                    this.refreshFunction(requestData);
+                    this.hide();
+                }
+            }.bind(this), { once: true }, false);
```

Comment:
I think you're missing code to remove the event listener if the modal is closed/cancelled without picking a component.

Reply:
Also if you use an arrow function

```js
window.addEventListener("message", (event) => {
```

then you can remove the `.bind(this)` as it won't be necessary.

Reply:
Like this: https://github.com/open-craft/edx-platform/commit/5939b5dada9f44f854d11dbe31ebef313349096a

Turns out this is actually necessary or sometimes a random message will come in and use up the "once" listener, and then the "Add" button will do nothing.

Reply:
Perfect!